integration/vm_test.go

```diff
@@ -0,0 +1,148 @@
+//go:build vm_integration
+
+package integration
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/aquasecurity/trivy/pkg/types"
+)
+
+func TestVM(t *testing.T) {
+	type args struct {
+		input        string
+		format       string
+		artifactType string
+	}
+	tests := []struct {
+		name     string
+		args     args
+		golden   string
+		override types.Report
+	}{
+		{
+			name: "amazon linux 2 in VMDK, filesystem XFS",
+			args: args{
+				input:        "testdata/fixtures/vm/amazonlinux2-gp2-x86.vmdk.img.gz",
+				format:       "json",
+				artifactType: "vm",
+			},
+			golden: "testdata/amazonlinux2-gp2-x86-vm.json.golden",
+		},
+		{
+			name: "amazon linux 2 in Snapshot, filesystem XFS",
+			args: args{
+				input:        "testdata/fixtures/vm/amazonlinux2-gp2-x86.img.gz",
+				format:       "json",
+				artifactType: "vm",
+			},
+			golden: "testdata/amazonlinux2-gp2-x86-vm.json.golden",
+		},
+		{
+			name: "RedHat in Snapshot, filesystem XFS",
+			args: args{
+				input:        "testdata/fixtures/vm/redhat-gp2-x86.img.gz",
+				format:       "json",
+				artifactType: "vm",
+			},
+			golden: "testdata/redhat-gp2-x86-vm.json.golden",
+		},
+		{
+			name: "RedHat in VMDK, filesystem XFS",
+			args: args{
+				input:        "testdata/fixtures/vm/redhat-gp2-x86.vmdk.img.gz",
+				format:       "json",
+				artifactType: "vm",
+			},
+			golden: "testdata/redhat-gp2-x86-vm.json.golden",
+		},
+		{
+			name: "SUSE in Snapshot, filesystem XFS",
+			args: args{
+				input:        "testdata/fixtures/vm/suse-gp2-x86.img.gz",
+				format:       "json",
+				artifactType: "vm",
+			},
+			golden: "testdata/suse-gp2-x86-vm.json.golden",
+		},
+		{
+			name: "Ubuntu in Snapshot, filesystem EXT4",
+			args: args{
+				input:        "testdata/fixtures/vm/ubuntu-gp2-x86.img.gz",
+				format:       "json",
+				artifactType: "vm",
+			},
+			golden: "testdata/ubuntu-gp2-x86-vm.json.golden",
+		},
+		{
+			name: "Ubuntu in VMDK, filesystem EXT4",
+			args: args{
+				input:        "testdata/fixtures/vm/ubuntu-gp2-x86.vmdk.img.gz",
+				format:       "json",
+				artifactType: "vm",
+			},
+			golden: "testdata/ubuntu-gp2-x86-vm.json.golden",
+		},
+	}
+
+	// Set up testing DB
+	cacheDir := initDB(t)
+
+	dirName := "./tmp-test-vm-integration-dir"
+	err := os.Mkdir(dirName, 0700)
```

Comment:
Because Target is include directory path. 


https://github.com/aquasecurity/trivy/pull/2910/files#diff-ba9f452540511ba6049047aea1faaeba2a76cbfec9d81b8c25d86de998f88fe3R23